### PR TITLE
Members area flow

### PR DESCRIPTION
**Pull request by davidrapson**
This PR is a combination of https://github.com/guardian/membership-frontend/pull/393 and https://github.com/guardian/membership-frontend/pull/366 with additional tests from @jennysivapalan 

- Updates to choose tier for members area content
- Shows an article snapshot on the joiner page if the user joined via members area content on gu.com.

# Choose Tier

screen shot 2015-03-24 at 16 51 20

screen shot 2015-03-24 at 16 51 40

# Joiner Thank you

**Content**
joiner - content

**Event**
joiner - event

**Default**
joiner - default

@jamesoram @jennysivapalan @rtyley

### CODE REVIEW

**Comment by jamesoram** [comment]
:+1:

File: frontend/app/services/DestinationService.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import com.netaporter.uri.dsl._
+import configuration.Config
+import model.{ContentDestination, Destination, EventDestination, MembersOnlyContent}
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+
+import scala.concurrent.Future
+
+trait DestinationService {
+
+  val JoinReferrer = "join-referrer"
```
Comment (mattandrews) [inline frontend/app/services/DestinationService.scala]: :warning: **=== boring style opinion follows ===** :warning: 

`JoinReferrer` should probably be `joinReferrer`, right?

Comment (mattandrews) [inline frontend/app/services/DestinationService.scala]: Actually, who knows... looking below, there's all kinds of casing going on. Ignore me!

File: frontend/app/services/DestinationService.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }.getOrElse(Future.successful(None))
+  }
+
+  def eventDestinationFor(request: AnyMemberTierRequest[_]): Future[Option[EventDestination]] = {
+    val optFuture = for {
+      eventId <- PreMembershipJoiningEventFromSessionExtractor.eventIdFrom(request)
+      event <- eventbriteService.getBookableEvent(eventId)
+    } yield memberService.createDiscountForMember(request.member, event).map { discountOpt =>
+        EventDestination(event, (Config.eventbriteApiIframeUrl ? ("eid" -> event.id) & ("discount" -> discountOpt.map(_.code))))
+      }
```
Comment (mattandrews) [inline frontend/app/services/DestinationService.scala]: This indentation is making my OCD hurt

File: frontend/app/services/DestinationService.scala (hunk 1 above)
Comment (davidrapson) [inline frontend/app/services/DestinationService.scala]: Yeah, I don't know either…

**Comment by mattandrews** [comment]
:+1:

File: frontend/app/services/DestinationService.scala (hunk 1 above)
Comment (rtyley) [inline frontend/app/services/DestinationService.scala]: It's probably right, if you're going by the official Scala style guide:

Comment (davidrapson) [inline frontend/app/services/DestinationService.scala]: In a PR this size we're talking about casing…: :bike: :house:

**Comment by davidrapson** [comment]
Confirmed on PROD in each scenario

- Fresh signup
- Signup via event
- Signup via guardian.com article

screen shot 2015-04-22 at 15 40 36
screen shot 2015-04-22 at 15 38 55
screen shot 2015-04-22 at 15 37 28

**Pull request merged**